### PR TITLE
Add cross entropy layer for pixel classification

### LINKED ISSUE #2875: Add cross entropy layer for pixel classification

### PR DESCRIPTION
**Pull request by wanghaoshuang**
1. Add pixel_cross_entropy_cost layer
2. Add INPUT_MULTI_LABEL type for generating pixel label data in LayerGradUtil.cpp
fix #2875

### CODE REVIEW

**Comment by qingqing01** [comment]
@wanghaoshuang as we discussed earlier, I still think there is no need to develop `pixel_cross_entropy_cost`. And this layer is not a general operation and a litter strange. We can expose the `switch_layer` for users and use the `switch_layer`,  `resize_layer`, `softmax` and `cross_entropy_cost` for the pixel classification:

```
switch_layer -> resize_layer -> softmax -> cross_entropy_cost
```

**Comment by wanghaoshuang** [comment]
@qingqing01 I implement switch_layer with reshape option in [PR#2788](https://github.com/PaddlePaddle/Paddle/pull/2788) which support for  softmax activation.
So i closed this PR.
